dotnetflix.Web/Pages/Checkout/CheckoutPage.razor

```diff
@@ -12,49 +14,103 @@
             <h3>Order Summary</h3>
             <table class="table">
                 <thead>
-                <tr>
-                    <th>Ticket Type</th>
-                    <th>Quantity</th>
-                    <th>Price per Ticket</th>
-                    <th>Total Price</th>
-                </tr>
+                    <tr>
+                        <th>Ticket Type</th>
+                        <th>Quantity</th>
+                        <th>Price per Ticket</th>
+                        <th>Total Price</th>
+                    </tr>
                 </thead>
                 <tbody>
-                @if (OrderRequest.TicketOrders.Any())
-                {
-                    decimal grandTotal = 0;
-                    foreach (var ticketOrder in OrderRequest.TicketOrders.Where(to => to.Quantity > 0))
+                    @if (OrderRequest.TicketOrders.Any())
                     {
-                        var ticketType = TicketTypes.FirstOrDefault(tt => tt.Id == ticketOrder.TicketTypeId);
-                        if (ticketType != null)
+                        decimal grandTotal = 0;
+                        foreach (var ticketOrder in OrderRequest.TicketOrders.Where(to => to.Quantity > 0))
                         {
-                            decimal pricePerTicket = ticketOrder.FinalPrice / ticketOrder.Quantity;
-                            grandTotal += ticketOrder.FinalPrice;
-
-                            <tr>
-                                <td>@ticketType.Name</td>
-                                <td>@ticketOrder.Quantity</td>
-                                <td>@pricePerTicket.ToString("C")</td>
-                                <td>@ticketOrder.FinalPrice.ToString("C")</td>
-                            </tr>
+                            var ticketType = TicketTypes.FirstOrDefault(tt => tt.Id == ticketOrder.TicketTypeId);
+                            if (ticketType != null)
+                            {
+                                decimal pricePerTicket = ticketOrder.FinalPrice / ticketOrder.Quantity;
+                                grandTotal += ticketOrder.FinalPrice;
+
+                                // Adjust grandTotal based on arrangements
```

Comment:
Waarom pas je hier nogmaals logica toe voor het berekenen van een prijs. Deze logica past beter in de OrderService, waar al logica bestaat voor het berekenen van ticketprices. 

Ook kun je dan een arrangement per ticket berekenen als dezelfde logica volgt.